packages/google-compute-engine-oslogin/README.md

```diff
@@ -1,76 +1,73 @@
-    profile and make them available to sshd.
-*   **NSS Module** provides support for making OS Login user and group
+*   **Authorized Keys Command** which provides SSH keys from the user's OS Login
+    profile to sshd for authenticating users at login.
+*   **NSS Modules** which provide support for making OS Login user and group
     information available to the system, using NSS (Name Service Switch)
     functionality.
-*   **PAM Module** provides authorization and authentication support allowing
-    the system to use data stored in Google Cloud IAM permissions to control
-    both, the ability to log into an instance, and to perform operations as root
-    (sudo).
-*   **Utils** provides common code to support the components listed above.
-
-In addition to the main components, there are also utilities for packaging and
-installing these components:
-
-*   **bin** contains a shell script for activating/deactivating the package
-    components.
-*   **packaging** contains files used to generate `.deb` and `.rpm` packages for
-    the OS Login components.
-*   **policy** contains SELinux "type enforcement" files for configuring SELinux
-    on CentOS/RHEL systems.
+*   **PAM Modules** which provide authorization (and authentication if
+    two-factor support is enabled) support allowing the system to use Google
+    Cloud IAM permissions to control the ability to log into an instance or to
+    perform operations as root (via `sudo`).
+
+In addition to the main components, there are also the following utilities:
+
+*   **google_oslogin_control** is a shell script for activating/deactivating the
+    OS Login components.
+*   **google_oslogin_nss_cache** is a utility for updating the local user and
+    group cache.
+*   **selinux** contains SELinux policy definition files and a compiled policy
+    package for configuring SELinux to support OS Login.
+
+The **packaging** directory also contains files used to generate `.deb` and
+`.rpm` packages for the OS Login components.
 
 ## Components
 
 #### Authorized Keys Command
 
 The `google_authorized_keys` binary is designed to be used with the sshd
-[AuthorizedKeysCommand](https://linux.die.net/man/5/sshd_config) option in
-`sshd_config`. It does the following:
+`AuthorizedKeysCommand` option in [sshd_config(5)](https://linux.die.net/man/5/sshd_config)
+It does the following:
 
-*   Reads the user's profile information from the metadata server.
+*   Reads the user's profile information from the metadata server:
     ```
     http://metadata.google.internal/computeMetadata/v1/oslogin/users?username=<username>
     ```
-*   Checks to make sure that the user is authorized to log in.
+*   Checks to make sure that the user is authorized to log in:
     ```
     http://metadata.google.internal/computeMetadata/v1/oslogin/authorize?email=<user_email>&policy=login
     ```
 *   If the check is successful, returns the SSH keys associated with the user
     for use by sshd.
```

Comment:
Might be useful to note that the authorized keys command will return a non-zero exit code if the authorization check fails.